lighthouse-core/computed/critical-request-chains.js

```diff
@@ -42,6 +43,8 @@ class CriticalRequestChains {
       NetworkRequest.TYPES.EventSource,
     ];
     if (nonCriticalResourceTypes.includes(request.resourceType || 'Other') ||
+        // if an iframe as a 302 it resourceType is undefined #6675
+        (!request.resourceType && !hasStatusCode200) ||
```

Comment:
I could omit the request.resourceType check this bugs only occured on iframes with a redirect so I like to be really explicit about it.

Reply:
Generally, restricting based on status code and missing resource type actually sounds pretty good to me (maybe just no 4xx/5xx in generic terms).

The only sticking point I'm coming up with this is if the root document request was a redirect, what's going to happen? I just realized this bug might be OOPIF related so if we're fairly confident this only happens to iframes we're probably OK.

Aside, I'm somewhat surprised we didn't include `'Other'` in the `nonCriticalResourceTypes` already which would stop this too, but a lot of things can wind up being `Other` I guess :/

Reply:
it's actually undefined :p

Reply:
@patrickhulce so resourceType is actually undefined when it's a 302. I'm now checking if redirectDestination exists and use it's resourceType. Maybe I should make it recursive? Now i only check one level